plugins/modules/win_domain_ou.ps1

```diff
@@ -0,0 +1,242 @@
+        $Properties.Add($_)
+    }
+    $extra_args.Properties = $Properties
+}else{
+    $extra_args.Properties = '*'
+    $Properties = '*'
+}
+
+$extra_args.Filter = $module.Params.filter
+$check_mode = $module.CheckMode
+$name = $module.Params.name
+$protected = $module.Params.protected
+$path = $module.Params.path
+$state = $module.Params.state
+$recursive = $module.Params.recursive
+
+# setup Dynamic Params
+$params = @{}
+if ($module.Params.properties.count -ne 0){
+    $module.Params.properties.Keys | ForEach-Object{
+        $params.Add($_,$module.Params.properties.Item($_))
+    }
+}
+# capture settings to know what we are doing
+$module.Result.params = $params
+$module.Result.extra_args = $extra_args
+$module.Result.onboard_extra_args = $onboard_extra_args
+$module.Result.properties = $properties
+
+Function Compare-OuObject {
+    Param(
+        [PSObject]$Original,
+        [PSObject]$Updated,
+        $properties
+    )
+    if (($Null -eq $Original) -or ($Original -eq $false)) { return $false }
+    if ($properties -ne '*'){
+        $x = Compare-Object -ReferenceObject $Original -DifferenceObject $Updated -Property $properties
+        #$module.Result.original = $Original | ConvertTo-Json -Compress
+        #$module.Result.updated = $Updated | ConvertTo-Json -Compress
+        #$module.Result.compare_properties = $properties
+        #$module.FailJson("Testing: failed compare $($x.count) $($properties.GetType().Name)")
+
+    }else{
+        $x = Compare-Object -ReferenceObject $Original -DifferenceObject $Updated
+    }
+    return $x.Count -eq 0
+}
+
+Function Get-SimulatedOu {
+    Param($Object)
+    $params = @{
+        Name = $Object.name
+        DistinguishedName = "OU=$($Object.name),$($Object.path)"
+        ProtectedFromAccidentalDeletion = $Object.protected
+    }
+    if ($Object.properties) {
+        if ($Object.properties.description) { $params.Description = $Object.properties.description }
+        if ($Object.properties.city) { $params.City = $Object.properties.city }
+        if ($Object.properties.state) { $params.State = $Object.properties.state }
+        if ($Object.properties.street_address) { $params.StreetAddress = $Object.properties.street_address }
+        if ($Object.properties.postal_code) { $params.PostalCode = $Object.properties.postal_code }
+        if ($Object.properties.country) { $params.Country = $Object.properties.country }
+        if ($Object.properties.managed_by) { $params.ManagedBy = $Object.properties.managed_by }
+    }
+    # convert to psobject & return
+    [PSCustomObject]$params
+}
+
+Function Get-OuObject {
+    Param([PSObject]$Object)
+    $obj = $Object | Select-Object -Property * -ExcludeProperty nTSecurityDescriptor | ConvertTo-Json -Depth 1 | ConvertFrom-Json
+    return $obj
+}
+
+# attempt import of module
+Try { Import-Module ActiveDirectory }
+Catch { $module.FailJson("Line 124: The ActiveDirectory module failed to load properly: $($_.Exception.Message)", $_) }
```

Comment:
Probably best to exclude `Line 124` in the error. `FailJson` will provide the stacktrace which includes the line numbers when you provide the error/exception which you have here. Ditto for the rest.